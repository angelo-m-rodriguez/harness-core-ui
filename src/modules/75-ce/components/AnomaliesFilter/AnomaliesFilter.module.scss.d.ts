/* eslint-disable */
/**
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 **/
// this is an auto-generated file, do not update this manually
declare const styles: {
  readonly cpFilters: string
  readonly cpLabel: string
  readonly drawerContent: string
  readonly exitIcon: string
  readonly expanded: string
  readonly filterCard: string
  readonly filterDrawer: string
  readonly formContainer: string
  readonly formHeader: string
  readonly header: string
  readonly label: string
  readonly multiSelect: string
  readonly newFilterBtn: string
  readonly pointer: string
  readonly popoverOption: string
  readonly radioBtnLabel: string
  readonly saveFilterContainer: string
  readonly savedFilter: string
  readonly selected: string
  readonly tagInput: string
}
export default styles
