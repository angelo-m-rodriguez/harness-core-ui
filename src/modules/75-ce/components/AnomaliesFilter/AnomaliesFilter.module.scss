/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.filterDrawer {
  .exitIcon {
    position: absolute;
    top: var(--spacing-xlarge);
    right: var(--spacing-xlarge);
    cursor: pointer;
  }

  .drawerContent {
    height: 100%;

    .formContainer {
      overflow: scroll;

      .label {
        padding-bottom: var(--spacing-small);
        color: var(--grey-700);

        &.cpLabel {
          padding-bottom: 0;
        }
      }

      .tagInput {
        box-shadow: none;
        border: 1px solid var(--grey-200);
        border-radius: var(--spacing-2);
        margin-bottom: var(--spacing-4);
      }

      .filterCard {
        padding: var(--spacing-small) var(--spacing-medium) !important;
        margin-bottom: var(--spacing-medium);
        margin-top: var(--spacing-small);

        .cpFilters {
          border-left: 1px solid var(--grey-300);
          padding-left: var(--spacing-medium);
          margin-top: var(--spacing-medium);
        }

        &.expanded {
          background-color: var(--primary-bg);
        }
      }

      .formHeader {
        padding-bottom: var(--spacing-small);
        border-bottom: 1px solid var(--grey-200);
      }

      .multiSelect {
        padding-top: var(--spacing-xsmall);
        padding-bottom: var(--spacing-medium);
      }
    }

    .saveFilterContainer {
      height: 100%;

      .newFilterBtn {
        padding: var(--spacing-medium) var(--spacing-xlarge);
        cursor: pointer;

        &:hover,
        &.selected {
          background-color: rgba(26, 26, 26, 0.4);
        }
      }

      .radioBtnLabel {
        color: var(--white) !important;
      }

      input[class*='bp3-input'] {
        color: var(--white) !important;
      }

      .savedFilter {
        padding: var(--spacing-medium) var(--spacing-xlarge);
        cursor: pointer;

        &:hover,
        &.selected {
          background-color: rgba(26, 26, 26, 0.4);
        }
      }
    }
  }
}

.popoverOption {
  cursor: pointer;
  padding: var(--spacing-medium) var(--spacing-large) !important;
}

.header {
  padding: var(--spacing-medium) var(--spacing-large) !important;
  border-bottom: 1px solid var(--grey-200);
  background: var(--white) !important;
}

.pointer {
  cursor: pointer;
}
