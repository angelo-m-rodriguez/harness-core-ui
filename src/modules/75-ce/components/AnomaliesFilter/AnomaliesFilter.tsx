/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { DropDown, FlexExpander, Layout, Text, Icon, SelectOption } from '@wings-software/uicore'
import { Color } from '@harness/design-system'
import { useStrings } from 'framework/strings'
import type { AnomalyFilterProperties, FilterStatsDTO } from 'services/ce'
import PerspectiveTimeRangePicker from '@ce/components/PerspectiveTimeRangePicker/PerspectiveTimeRangePicker'
import type { setTimeRangeFn } from '@ce/types'
import useAnomaliesFilterPanel from './AnomaliesFilterPanel'

import css from './AnomaliesFilter.module.scss'

interface AnomalyFiltersProps {
  filters: AnomalyFilterProperties
  setFilters: (newValue: AnomalyFilterProperties) => void
  timeRange: {
    to: string
    from: string
  }
  setTimeRange: setTimeRangeFn
  fetchedFilterValues: FilterStatsDTO[]
  fetching: boolean
}

const AnomalyFilters: React.FC<AnomalyFiltersProps> = ({
  filters,
  setFilters,
  timeRange,
  setTimeRange,
  fetching,
  fetchedFilterValues
}) => {
  const { getString } = useStrings()

  const { openFilterDrawer, savedFilters, setSavedFilter, selectedFilter } = useAnomaliesFilterPanel({
    fetchedFilterValues,
    filters,
    setFilters
  })

  const savedFilterOptions = savedFilters.map(filter => ({
    label: filter.name,
    value: filter.identifier
  })) as SelectOption[]

  return (
    <Layout.Horizontal spacing="large" className={css.header}>
      <Layout.Horizontal spacing="large" style={{ alignItems: 'center' }}></Layout.Horizontal>
      <FlexExpander />
      {fetching ? (
        <Icon name="spinner" color={Color.BLUE_500} size={24} />
      ) : (
        <>
          <DropDown
            placeholder={getString(savedFilters.length > 0 ? 'filters.selectFilter' : 'common.filters.noFilterSaved')}
            filterable={false}
            disabled={savedFilters.length === 0}
            onChange={option => setSavedFilter(option)}
            addClearBtn={true}
            value={selectedFilter?.identifier}
            items={savedFilterOptions}
          />
          <Icon
            name="ng-filter"
            size={24}
            onClick={openFilterDrawer}
            color={Color.PRIMARY_7}
            className={css.pointer}
            style={{ alignSelf: 'center' }}
          />
          <Text border={{ right: true, color: Color.GREY_300 }} />
          <PerspectiveTimeRangePicker timeRange={timeRange} setTimeRange={setTimeRange} />
        </>
      )}
    </Layout.Horizontal>
  )
}

export default AnomalyFilters
