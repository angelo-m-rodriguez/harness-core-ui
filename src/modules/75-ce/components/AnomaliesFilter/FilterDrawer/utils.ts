/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MultiSelectOption } from '@harness/uicore'
import { omit } from 'lodash-es'
import type { AnomalyFilterProperties } from 'services/ce'
import type { AnomaliesFilterFormType } from './FilterDrawer'

const getMultiSelectOptions = (values: string[]): MultiSelectOption[] =>
  values.map(item => ({ label: item, value: item }))

const getValueFromOption = (values: MultiSelectOption[]): string[] => values.map(item => item.value.toString())

export const getAnomalyFormValuesFromFilterProperties = (
  filterProperties: AnomalyFilterProperties
): AnomaliesFilterFormType => {
  const formValues: AnomaliesFilterFormType = {}

  Object.entries(omit(filterProperties, 'filterType')).forEach(([key, value]) => {
    if (value) {
      if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
        formValues[key] = +value
      } else {
        ;(formValues as any)[key] = getMultiSelectOptions(value as string[])
      }
    }
  })

  return formValues
}

export const getAnomalyFilterPropertiesFromForm = (formData: AnomaliesFilterFormType): AnomalyFilterProperties => {
  const filterProperties: AnomalyFilterProperties = { filterType: 'Anomaly' }

  Object.entries(formData).forEach(([key, value]) => {
    if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
      filterProperties[key] = +value
    } else {
      ;(filterProperties as any)[key] = getValueFromOption(value)
    }
  })

  return filterProperties
}
