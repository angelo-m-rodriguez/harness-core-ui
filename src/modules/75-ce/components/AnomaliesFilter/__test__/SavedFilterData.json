{
  "data": {
    "totalPages": 1,
    "totalItems": 4,
    "pageItemCount": 4,
    "pageSize": 100,
    "content": [
      {
        "name": "Test1",
        "identifier": "Test1",
        "orgIdentifier": null,
        "projectIdentifier": null,
        "filterProperties": {
          "k8sClusterNames": ["sample-ce-dev"],
          "k8sNamespaces": null,
          "k8sWorkloadNames": null,
          "gcpProjects": null,
          "gcpProducts": null,
          "gcpSKUDescriptions": null,
          "awsAccounts": null,
          "awsServices": null,
          "awsUsageTypes": null,
          "azureSubscriptions": null,
          "azureServiceNames": null,
          "azureResources": null,
          "minActualAmount": null,
          "minAnomalousSpend": null,
          "timeFilters": null,
          "offset": null,
          "limit": null,
          "tags": {
            "1": "",
            "22": ""
          },
          "filterType": "Anomaly"
        },
        "filterVisibility": "EveryOne"
      },
      {
        "name": "Test_3",
        "identifier": "Test_3",
        "orgIdentifier": null,
        "projectIdentifier": null,
        "filterProperties": {
          "k8sClusterNames": null,
          "k8sNamespaces": null,
          "k8sWorkloadNames": null,
          "gcpProjects": null,
          "gcpProducts": null,
          "gcpSKUDescriptions": null,
          "awsAccounts": null,
          "awsServices": null,
          "awsUsageTypes": null,
          "azureSubscriptions": null,
          "azureServiceNames": null,
          "azureResources": null,
          "minActualAmount": null,
          "minAnomalousSpend": null,
          "timeFilters": null,
          "offset": null,
          "limit": null,
          "tags": {},
          "filterType": "Anomaly"
        },
        "filterVisibility": "OnlyCreator"
      }
    ],
    "pageIndex": 0,
    "empty": false
  }
}
