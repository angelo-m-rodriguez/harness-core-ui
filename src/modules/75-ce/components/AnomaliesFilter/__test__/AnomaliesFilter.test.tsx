/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import {
  queryByText,
  render,
  fireEvent,
  act,
  getByText,
  getByPlaceholderText,
  queryByPlaceholderText,
  waitFor,
  getAllByPlaceholderText
} from '@testing-library/react'
import { findPopoverContainer, TestWrapper } from '@common/utils/testUtils'

import AnomaliesFilter from '../AnomaliesFilter'
import SavedFilterData from './SavedFilterData.json'

const findDrawerContainer = (): HTMLElement | null => document.querySelector('.bp3-drawer')

jest.mock('services/ce', () => ({
  useGetFilterList: jest
    .fn()
    .mockImplementationOnce(() => {
      return {
        data: {},
        refetch: jest.fn(),
        loading: false
      }
    })
    .mockImplementation(() => {
      return {
        data: SavedFilterData,
        refetch: jest.fn(),
        loading: false
      }
    }),
  usePostFilter: jest.fn().mockImplementation(() => {
    return {
      mutate: jest.fn()
    }
  }),
  useUpdateFilter: jest.fn().mockImplementation(() => {
    return {
      mutate: jest.fn()
    }
  }),
  useDeleteFilter: jest.fn().mockImplementation(() => {
    return {
      mutate: jest.fn()
    }
  })
}))

const params = {
  accountId: 'TEST_ACC',
  perspetiveId: 'perspectiveId',
  perspectiveName: 'sample perspective'
}

describe('test case for anomalies detection overview page', () => {
  test('Should be able to render component when no saved filters', () => {
    const setFilters = jest.fn()
    const { container } = render(
      <TestWrapper pathParams={params}>
        <AnomaliesFilter
          filters={{}}
          setFilters={setFilters}
          timeRange={{
            to: '2022-10-02',
            from: '2022-10-02'
          }}
          setTimeRange={jest.fn}
          fetchedFilterValues={[]}
          fetching={false}
        />
      </TestWrapper>
    )

    expect(queryByText(container, 'common.filters.noFilterSaved')).toBeDefined()
  })

  test('should be able to render the anomaly filters when loading', async () => {
    const setFilters = jest.fn()
    const { container } = render(
      <TestWrapper pathParams={params}>
        <AnomaliesFilter
          filters={{}}
          setFilters={setFilters}
          timeRange={{
            to: '2022-10-02',
            from: '2022-10-02'
          }}
          setTimeRange={jest.fn}
          fetchedFilterValues={[{ key: 'key1', values: ['val1'] }]}
          fetching={true}
        />
      </TestWrapper>
    )

    expect(container.querySelector('[data-icon="spinner"]')).toBeDefined()
  })

  const props = {
    fetching: false,
    filters: { k8sClusterNames: ['sample-ce-dev'], awsAccounts: ['ac1'], tags: { tag1: '', TAG2: '' } },
    setFilters: jest.fn(),
    setTimeRange: jest.fn(),
    fetchedFilterValues: [
      { key: 'key1', values: ['val1'] },
      { key: 'clustername', values: ['sample-ce-dev', 'name_2'] },
      { key: 'awsaccount', values: ['ac1', 'acc2'] },
      { key: 'gcpproduct', values: ['pd1', 'pd2'] },
      { key: 'namespace', values: ['name1', 'name2'] }
    ],
    timeRange: { to: '2022-10-02', from: '2022-10-02' }
  }

  test('should be able to edit and delete saved filter', async () => {
    const { container } = render(
      <TestWrapper pathParams={params}>
        <AnomaliesFilter {...props} />
      </TestWrapper>
    )

    expect(queryByText(container, 'filters.selectFilter')).toBeDefined()

    const filterPanelButton = container.querySelector('[data-icon="ng-filter"]')

    act(() => {
      fireEvent.click(filterPanelButton!)
    })

    const drawer = findDrawerContainer()

    expect(getByText(drawer!, 'ce.anomalyDetection.filters.anomalousSpend')).toBeDefined()
    fireEvent.click(drawer?.querySelector('[icon="plus"]')!)
    expect(getByPlaceholderText(drawer!, 'filters.typeFilterName')).toBeDefined()
    fireEvent.click(queryByText(drawer!, 'Test1')!)
    fireEvent.click(drawer?.querySelectorAll('[data-icon="chevron-down"]')[0]!)
    fireEvent.click(drawer?.querySelectorAll('[data-icon="Options"]')[0]!)

    const menu1 = findPopoverContainer()
    fireEvent.click(menu1?.querySelector('[data-icon="edit"]')!)
    fireEvent.click(getByText(drawer!, 'update'))

    fireEvent.click(queryByText(drawer!, 'Test_3')!)
    fireEvent.click(drawer?.querySelectorAll('[data-icon="Options"]')[1]!)
    const menu2 = findPopoverContainer()
    fireEvent.click(menu2?.querySelector('[data-icon="trash"]')!)
  })

  test('Should be able to create new filter', async () => {
    const { container } = render(
      <TestWrapper pathParams={params}>
        <AnomaliesFilter {...props} />
      </TestWrapper>
    )

    expect(queryByText(container, 'filters.selectFilter')).toBeDefined()

    const filterPanelButton = container.querySelector('[data-icon="ng-filter"]')

    act(() => {
      fireEvent.click(filterPanelButton!)
    })

    const drawer = findDrawerContainer()
    fireEvent.click(drawer?.querySelector('[icon="plus"]')!)

    expect(getByPlaceholderText(drawer!, 'filters.typeFilterName')).toBeDefined()

    fireEvent.change(getByPlaceholderText(drawer!, 'filters.typeFilterName'), { target: { value: 'mock_name' } })
    fireEvent.click(drawer?.querySelector('input[value="OnlyCreator"]')!)
    fireEvent.click(drawer?.querySelector('input[value="EveryOne"]')!)
    fireEvent.click(getByText(drawer!, 'save'))

    await waitFor(() => expect(queryByPlaceholderText(drawer!, 'filters.typeFilterName')).toBeNull())
  })

  test('Should be able to select saved filter', () => {
    const setFilters = jest.fn()

    const { container } = render(
      <TestWrapper pathParams={params}>
        <AnomaliesFilter {...props} setFilters={setFilters} />
      </TestWrapper>
    )

    const selectedFilter = queryByText(container, 'filters.selectFilter')

    fireEvent.click(selectedFilter!)

    fireEvent.click(queryByText(container, 'Test_3')!)

    expect(setFilters).toBeCalledWith(SavedFilterData.data.content[1].filterProperties)
  })

  test('Should be able to set and apply filters and clear them', async () => {
    const { container } = render(
      <TestWrapper pathParams={params}>
        <AnomaliesFilter {...props} />
      </TestWrapper>
    )

    const filterPanelButton = container.querySelector('[data-icon="ng-filter"]')

    act(() => {
      fireEvent.click(filterPanelButton!)
    })

    const drawer = findDrawerContainer()

    fireEvent.change(getAllByPlaceholderText(drawer!, 'ce.anomalyDetection.filters.spendPlaceholder')[0]!, {
      target: { value: 200 }
    })
    fireEvent.click(getByText(drawer!, 'filters.clearAll'))

    fireEvent.click(drawer?.querySelectorAll('[data-icon="chevron-down"]')[0]!)
    fireEvent.click(drawer?.querySelectorAll('[data-icon="chevron-down"]')[0]!)
    fireEvent.click(drawer?.querySelectorAll('[data-icon="chevron-down"]')[0]!)

    fireEvent.change(getAllByPlaceholderText(drawer!, 'ce.anomalyDetection.filters.spendPlaceholder')[1]!, {
      target: { value: 150 }
    })
    fireEvent.click(getByText(drawer!, 'common.apply'))

    expect(drawer?.querySelector('input[value="150"]')).toBeDefined()
  })
})
