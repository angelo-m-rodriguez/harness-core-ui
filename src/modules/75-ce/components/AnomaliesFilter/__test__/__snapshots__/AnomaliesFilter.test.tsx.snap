// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test case for anomalies detection overview page Should be able to create new filter 1`] = `
<div
  class="bp3-drawer bp3-position-right filterDrawer bp3-overlay-content bp3-overlay-appear bp3-overlay-appear-active"
  style="width: 632px;"
  tabindex="0"
>
  <div
    class="Layout--horizontal StyledProps--main drawerContent"
  >
    <span
      class="bp3-icon bp3-icon-cross StyledProps--main exitIcon StyledProps--color StyledProps--color-white"
      icon="cross"
    >
      <svg
        data-icon="cross"
        height="22"
        viewBox="0 0 20 20"
        width="22"
      >
        <desc>
          cross
        </desc>
        <path
          d="M11.41 10l4.29-4.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L10 8.59l-4.29-4.3a1.003 1.003 0 00-1.42 1.42L8.59 10 4.3 14.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4.29-4.3 4.29 4.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L11.41 10z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="Layout--vertical StyledProps--main formContainer StyledProps--padding-xlarge"
      style="width: 398px;"
    >
      <div
        class="StyledProps--font StyledProps--main formHeader"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-h5 StyledProps--color StyledProps--color-grey300"
        >
          filters.newFilter
        </p>
      </div>
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-top-xlarge"
        style="height: 100%;"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--padding-bottom-small StyledProps--color StyledProps--color-grey700 StyledProps--font-variation-small-semi"
        >
          tagsLabel
        </p>
        <div
          class="bp3-input bp3-tag-input tagInput"
        >
          <div
            class="bp3-tag-input-values"
          >
            <input
              class="bp3-input-ghost"
              value=""
            />
          </div>
        </div>
        <p
          class="StyledProps--font StyledProps--main StyledProps--padding-bottom-small StyledProps--color StyledProps--color-grey700 StyledProps--font-variation-small-semi"
        >
          ce.anomalyDetection.filters.actualSpend
        </p>
        <div
          class="TextInput--main"
        >
          <div
            class="bp3-input-group"
          >
            <input
              class="bp3-input"
              type="text"
              value="0"
            />
          </div>
        </div>
        <p
          class="StyledProps--font StyledProps--main StyledProps--padding-bottom-small StyledProps--color StyledProps--color-grey700 StyledProps--font-variation-small-semi"
        >
          ce.anomalyDetection.filters.anomalousSpend
        </p>
        <div
          class="TextInput--main"
        >
          <div
            class="bp3-input-group"
          >
            <input
              class="bp3-input"
              type="text"
              value="0"
            />
          </div>
        </div>
        <div
          class="bp3-card bp3-elevation-0 Card--card filterCard"
          style="width: 100%;"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700"
            >
              ce.anomalyDetection.filters.awsFilters
            </p>
            <span
              class="bp3-icon bp3-icon-chevron-down StyledProps--main pointer StyledProps--color StyledProps--color-primary7"
              icon="chevron-down"
            >
              <svg
                data-icon="chevron-down"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  chevron-down
                </desc>
                <path
                  d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="bp3-card bp3-elevation-0 Card--card filterCard"
          style="width: 100%;"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700"
            >
              ce.anomalyDetection.filters.clusterFilters
            </p>
            <span
              class="bp3-icon bp3-icon-chevron-down StyledProps--main pointer StyledProps--color StyledProps--color-primary7"
              icon="chevron-down"
            >
              <svg
                data-icon="chevron-down"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  chevron-down
                </desc>
                <path
                  d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="bp3-card bp3-elevation-0 Card--card filterCard"
          style="width: 100%;"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700"
            >
              ce.anomalyDetection.filters.gcpFilters
            </p>
            <span
              class="bp3-icon bp3-icon-chevron-down StyledProps--main pointer StyledProps--color StyledProps--color-primary7"
              icon="chevron-down"
            >
              <svg
                data-icon="chevron-down"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  chevron-down
                </desc>
                <path
                  d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main"
      >
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-primary"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            common.apply
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--margin-left-small Button--with-current-color Button--variation Button--variation-tertiary"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            filters.clearAll
          </span>
        </button>
      </div>
    </div>
    <div
      class="Layout--vertical StyledProps--main saveFilterContainer StyledProps--background-primary9"
      style="width: 234px;"
    >
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-xlarge"
      >
        <span
          class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-white"
          data-icon="ng-filter"
        >
          <svg
            height="22"
            viewBox="0 0 24 24"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M16.5 8V7H19a.5.5 0 000-1h-2.5V5a.5.5 0 00-1 0v1H5a.5.5 0 000 1h10.5v1a.5.5 0 001 0zM5 11.5h3v-1a.5.5 0 011 0v1h10a.5.5 0 010 1H9v1a.5.5 0 01-1 0v-1H5a.5.5 0 010-1zM5 17a.5.5 0 000 1h7v1a.5.5 0 001 0v-1h6a.5.5 0 000-1h-6v-1a.5.5 0 00-1 0v1H5z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-h5 StyledProps--color StyledProps--color-white StyledProps--margin-left-medium"
        >
          filters.filtersLabel
        </span>
      </div>
      <div
        class="StyledProps--font StyledProps--main newFilterBtn selected"
      >
        <span
          class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--color StyledProps--color-white"
          icon="plus"
        >
          <svg
            data-icon="plus"
            height="18"
            viewBox="0 0 16 16"
            width="18"
          >
            <desc>
              plus
            </desc>
            <path
              d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-small-semi StyledProps--color StyledProps--color-white StyledProps--margin-left-medium"
        >
          filters.newFilter
        </span>
      </div>
      <div
        class="StyledProps--font StyledProps--main"
        style="height: 100%;"
      >
        <div
          class="Layout--horizontal StyledProps--main savedFilter StyledProps--flex"
        >
          <div
            class="Layout--horizontal StyledProps--main"
          >
            <span
              class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--padding-left-xlarge StyledProps--font-variation-small-semi StyledProps--color StyledProps--color-white"
            >
              Test1
            </span>
          </div>
          <span
            class="bp3-popover-wrapper bp3-dark"
          >
            <span
              class="bp3-popover-target"
            >
              <span
                class="bp3-icon StyledProps--main"
                data-icon="Options"
              >
                <svg
                  fill="none"
                  height="16"
                  viewBox="0 0 4 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M2 12.5a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zm2-12a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zM2 7a1 1 0 100 2 1 1 0 000-2zM0 8a2 2 0 114 0 2 2 0 01-4 0z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </span>
        </div>
        <div
          class="Layout--horizontal StyledProps--main savedFilter StyledProps--flex"
        >
          <div
            class="Layout--horizontal StyledProps--main"
          >
            <span
              class="bp3-icon bp3-icon-lock StyledProps--main StyledProps--inline"
              icon="lock"
            >
              <svg
                data-icon="lock"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  lock
                </desc>
                <path
                  d="M13.96 7H12V3.95C12 1.77 10.21 0 8 0S4 1.77 4 3.95V7H1.96c-.55 0-.96.35-.96.9v6.91c0 .54.41 1.19.96 1.19h12c.55 0 1.04-.65 1.04-1.19V7.9c0-.55-.49-.9-1.04-.9zM6 7V3.95c0-1.09.9-1.97 2-1.97s2 .88 2 1.97V7H6z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--padding-left-small StyledProps--font-variation-small-semi StyledProps--color StyledProps--color-white"
            >
              Test_3
            </span>
          </div>
          <span
            class="bp3-popover-wrapper bp3-dark"
          >
            <span
              class="bp3-popover-target"
            >
              <span
                class="bp3-icon StyledProps--main"
                data-icon="Options"
              >
                <svg
                  fill="none"
                  height="16"
                  viewBox="0 0 4 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M2 12.5a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zm2-12a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zM2 7a1 1 0 100 2 1 1 0 000-2zM0 8a2 2 0 114 0 2 2 0 01-4 0z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </span>
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-xlarge"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--border StyledProps--border-top StyledProps--border-color-white StyledProps--padding-bottom-medium"
          style="opacity: 0.1;"
        />
        <p
          class="StyledProps--font StyledProps--main StyledProps--margin-bottom-medium StyledProps--color StyledProps--color-white"
        >
          filters.name
        </p>
        <div
          class="TextInput--main"
        >
          <div
            class="bp3-input-group"
          >
            <input
              class="bp3-input"
              color="white"
              placeholder="filters.typeFilterName"
              style="background-color: transparent;"
              type="text"
              value="mock_name"
            />
          </div>
        </div>
        <div
          class="Layout--vertical Layout--layout-spacing-small StyledProps--main StyledProps--margin-top-small StyledProps--margin-bottom-large"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-white StyledProps--margin-top-medium StyledProps--margin-bottom-medium"
          >
            filters.filterVisibility
          </p>
          <label
            class="bp3-control bp3-radio bp3-large StyledProps--font Radio--radio radioBtnLabel StyledProps--main"
          >
            <input
              checked=""
              type="radio"
              value="EveryOne"
            />
            <span
              class="bp3-control-indicator"
            />
            filters.visibleToEveryone
          </label>
          <label
            class="bp3-control bp3-radio bp3-large StyledProps--font Radio--radio radioBtnLabel StyledProps--main"
          >
            <input
              type="radio"
              value="OnlyCreator"
            />
            <span
              class="bp3-control-indicator"
            />
            filters.visibileToOnlyMe
          </label>
        </div>
        <button
          class="bp3-button bp3-disabled bp3-loading Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-secondary"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <div
            class="bp3-spinner bp3-button-spinner"
          >
            <div
              class="bp3-spinner-animation"
            >
              <svg
                height="20"
                stroke-width="16.00"
                viewBox="-3.00 -3.00 106.00 106.00"
                width="20"
              >
                <path
                  class="bp3-spinner-track"
                  d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                />
                <path
                  class="bp3-spinner-head"
                  d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                  pathLength="280"
                  stroke-dasharray="280 280"
                  stroke-dashoffset="210"
                />
              </svg>
            </div>
          </div>
          <span
            class="bp3-button-text"
          >
            save
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main radioBtnLabel StyledProps--margin-left-small Button--with-current-color Button--variation Button--variation-link"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            cancel
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
